sample_apps/rust/examples/utils/query_common.rs

```diff
@@ -0,0 +1,252 @@
+    // For example, query_results.log
+    #[arg(short, long, default_value = DEFAULT_OUTPUT_FILE)]
+    pub output_file: String,
+
+    // The name of the AWS region to use for queries
+    #[arg(short, long, default_value = DEFAULT_REGION)]
+    pub region: String,
+
+    // The Timestream for LiveAnalytics table name to use for all queries
+    #[arg(short, long, default_value = DEFAULT_TABLE_NAME)]
+    pub table_name: String,
+}
+
+pub async fn get_connection(
+    region: &String,
+) -> Result<timestream_query::Client, timestream_query::Error> {
+    let config = aws_config::defaults(aws_config::BehaviorVersion::latest())
+        .region(Region::new(region.to_owned()))
+        .load()
+        .await;
+    let (client, reload) = timestream_query::Client::new(&config)
+        .with_endpoint_discovery_enabled()
+        .await
+        .expect("Failure");
+    tokio::task::spawn(reload.reload_task());
+    Ok(client)
+}
+
+pub fn write(mut file: &fs::File, s: String) -> io::Result<()> {
+    let s_formatted = format!("{}\n", s);
+    file.write(s_formatted.as_bytes())?;
+    file.flush()?;
+    Ok(())
+}
+
+pub fn process_scalar_type(data: &types::Datum) -> Result<String, String> {
+    data.scalar_value
+        .clone()
+        .ok_or_else(|| "Scalar value is None".to_string())
+}
+
+pub fn process_time_series_type(
+    data: &[types::TimeSeriesDataPoint],
+    column_info: &types::ColumnInfo,
+) -> Result<String, String> {
+    let mut value = String::new();
+    for (i, datum) in data.iter().enumerate() {
+        value.push_str(&datum.time);
+        value.push(':');
+
+        let column_type = column_info.r#type();
+        let column_type_ref = column_type.as_ref().ok_or("Column type is None")?;
+        let scalar_type = column_type_ref.scalar_type.to_owned();
+        let scalar_type_ref = scalar_type.ok_or("Scalar type is None")?;
+
+        let datum_value = datum.value.as_ref().ok_or("Datum value is None")?;
+        let _scalar_value = datum_value
+            .scalar_value
+            .to_owned()
+            .ok_or("Scalar value is None")?;
+
+        if scalar_type_ref.as_str() == "" {
+            value.push_str(&process_scalar_type(datum_value)?);
+        } else if let Some(array_column_info) = &column_type_ref.array_column_info {
+            let array_value = datum_value
+                .array_value
+                .as_ref()
+                .ok_or("Array value is None")?;
+            value.push_str(&process_array_type(array_value, array_column_info)?);
+        } else if let Some(row_column_info) = &column_type_ref.row_column_info {
+            let row_value = datum_value.row_value.as_ref().ok_or("Row value is None")?;
+            value.push_str(&process_row_type(&row_value.data, row_column_info)?);
+        } else {
+            panic!("Bad data type");
+        }
+
+        if i != data.len() - 1 {
+            value.push_str(", ");
+        }
+    }
+    return Ok(value);
```

Comment:
Let's keep the returns consistent and avoid the unneeded bacon warnings.

```suggestion
   Ok(value);
```

Reply:
What bacon settings are you using? I am not getting any of the errors or warnings you've mentioned with `bacon check-all`.

Reply:
`bacon clippy -- --example query_sample`

Reply:
Changed to `Ok(value)`. Implicit returns are used for all other functions where possible.